isisd/isis_lsp.c

```diff
@@ -1164,6 +1164,26 @@ static void lsp_build(struct isis_lsp *lsp, struct isis_area *area)
 					 false, false, false);
 	}
 
+	/* Check if device should advertise high metrics on startup */
+	if (device_startup) {
+		uint32_t max_metric;
+		max_metric = isis_restart_read_advertise_high_metric(area);
+		if (max_metric > 0) {
+			isis_area_advertise_high_metrics_set(area, true,
+							     max_metric);
+		}
+
+		if (max_metric == ISIS_NARROW_METRIC_INFINITY) {
+			area->oldmetric = 1;
+		} else if (max_metric == MAX_NARROW_LINK_METRIC) {
+			area->oldmetric = 1;
+			area->newmetric = 0;
+		}
+
+		device_startup = false;
```

Comment:
`if (device_startup)` is also present in lsp_generate , which is called by this function.
Set to false will avoid setting the overload bit. Is it something that you want ?

Why `if (device_startup)` is present in two functions ? Could we move the overload bit and hight metric code in lsp_build or lsp_generate ?

Reply:
Removed `if (device_startup)` for this feature, will remove it for overload_on_startup in separate PR